compiler/rustc_typeck/src/collect.rs

```diff
@@ -254,10 +254,15 @@ impl Visitor<'tcx> for CollectItemTypesVisitor<'tcx> {
                     self.tcx.ensure().type_of(def_id);
                 }
                 hir::GenericParamKind::Type { .. } => {}
-                hir::GenericParamKind::Const { .. } => {
+                hir::GenericParamKind::Const { default, .. } => {
                     let def_id = self.tcx.hir().local_def_id(param.hir_id);
                     self.tcx.ensure().type_of(def_id);
-                    // FIXME(const_generics_defaults)
+                    if let Some(default) = default {
+                        let def_id = self.tcx.hir().local_def_id(default.hir_id);
+                        // need to store default and type of default
+                        self.tcx.ensure().type_of(def_id);
+                        self.tcx.ensure().const_param_default(def_id);
```

Comment:
```suggestion
                        let default_def_id = self.tcx.hir().local_def_id(default.hir_id);
                        // need to store default and type of default
                        self.tcx.ensure().type_of(default_def_id);
                        self.tcx.ensure().const_param_default(def_id);
```
`const_param_default` should be called for the param, not the default